### PR TITLE
fix: fixed corner cases for attempt restrictions update user story

### PR DESCRIPTION
**Pull request by martinwang-d2l**
Fixed issue related to story: https://rally1.rallydev.com/#/29180338367d/custom/486232622040?detail=%2Fuserstory%2F613198151415

Issue was distinguishing between whether a user is **able** to update a submission view's attempt restrictions, and whether the user **wants** to. In practice there are a few corner cases that this affects - where after unticking 'attempts' and reloading quiz, 'attempts' gets reticked again.

- Previously, attemptRestrictions on its own determines whether to show the attempt restrictions tickbox. A check in attempt restrictions editor determines whether attemptRestrictions is patched to true/false, upon a change to the quiz's attempt restriction number. 
- However, there exists the corner case where the user is able to update a view but does not want to (e.g. quiz attempt restriction = 5, view attempt restriction = 3). The current implementation checks "attempts" by default. 
- There also existed a corner case where the updated submission view store was not fetched. 

Fix: 

- Moved check in attempts function call to the container, **after** the checkin. The function call is still needed because the tile needs to have the latest submission view store. Now the call only does a fetch(), and does not patch attemptRestrictions, the restriction number comparison is also moved. 
- Introduced a local variable in access restrictions, this.canShow, which replaces the functionality of attemptRestrictions. Now, showing the attempt restrictions checkbox is conditional on `attemptRestrictions && this.canShow`

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js [hunk 1]
```diff
@@ -44,9 +44,17 @@ class ActivityQuizSubmissionViewReleaseDescription
...
 			timeLimit,
 			timeLimitNumber,
-			attemptRestrictionNumber
+			attemptRestrictionsOptions, // quiz attempts allowed.
+			attemptRestrictionNumber  // currently selected option.
 		} = view;
 
+		let canShow = false;
+		if (attemptRestrictions) {
+			if (attemptRestrictionNumber <= attemptRestrictionsOptions.length) {
```
[the remark this answers is not in the document]
Comment (mostlyfabulous) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js:53]: Provided `attemptRestrictionsOptions` always exists when `attemptRestrictions` is `true`

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js [hunk 2]
```diff
@@ -224,7 +233,9 @@ class ActivityQuizSubmissionViewsAccessRestrictionsEditor
 		if (!view) return;
 		view && view.setAttemptRestrictions(e.target.checked);
 
-		if (e.target.checked) {
+		// Override - want to make changes to unticked box.
```
Comment (mostlyfabulous) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js:236]: Could we edit the comment to elaborate more? I'm guessing the override is to update `canShow` manually so the UI reflects a user's choice to tick the attempt restrictions box.

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-attempts-container.js [hunk 3]
```diff
@@ -131,7 +146,6 @@ class ActivityQuizManageAttemptsContainer extends ActivityEditorWorkingCopyDialo
 		return html`
 			<d2l-activity-quiz-manage-attempts-editor
 				href="${attemptsHref}"
-				.quizHref="${this.checkedOutHref}"
```
Comment (mostlyfabulous) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-attempts-container.js:134]: Unused right?

Comment (martinwang-d2l) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-attempts-container.js:134]: Yep - I moved my update function from attempts editor to attempts container, so the HREF is no longer needed in the editor

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js (hunk 2 above)
Comment (martinwang-d2l) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js]: Added comment

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinwang-d2l) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js]: Yep, it should always exist

[the remark this answers is not in the document]
Comment (martinwang-d2l) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js]: I added a check, just to be safe though.

**Review by mostlyfabulous — APPROVED** [review]

**Pull request merged**